File: examples/FSharp/HelloWorld/HelloWorldExample.fs
module FSharp.HelloWorld.HelloWorldExample

open System.Threading.Tasks

open FSharp.Control.Tasks.V2.ContextInsensitive

open NBomber.Contracts
open NBomber.FSharp

let run () =

    let step1 = Step.create("step_1", fun context -> task {

        // you can do any logic here: go to http, websocket etc
        do! Task.Delay(seconds 1)
        return Response.Ok(42) // this value will be passed as response for the next step
    })

    let pause = Step.createPause(milliseconds 100)

    let step2 = Step.create("step_2", fun context -> task {
        let value = context.GetPreviousStepResponse<int>(); // 42
        return Response.Ok();
    })

    Scenario.create "hello_world_scenario" [step1; pause; step2]
    |> Scenario.withoutWarmUp
    |> Scenario.withLoadSimulations [KeepConstant(copies = 1, during = seconds 30)]
    |> NBomberRunner.registerScenario
    |> NBomberRunner.withTestSuite "example"
    |> NBomberRunner.withTestName "hello_world_test"
    |> NBomberRunner.run
    |> ignore

File: examples/CSharpDev/HelloWorld/HelloWorldExample.cs
using System;
using System.Threading.Tasks;
using NBomber.CSharp;

namespace CSharpDev.HelloWorld;

public class HelloWorldExample
{
    public void Run()
    {
        var scenario = Scenario.Create("hello_world_scenario", async context =>
        {
            // you can define and execute any logic here,
            // for example: send http request, SQL query etc
            // NBomber will measure how much time it takes to execute your logic
            await Task.Delay(1000);

            return Response.Ok();
        })
        .WithoutWarmUp()
        .WithLoadSimulations(
            Simulation.RampConstant(copies: 50, during: TimeSpan.FromMinutes(1)),
            Simulation.KeepConstant(copies: 50, during: TimeSpan.FromSeconds(30))
        );

        NBomberRunner
            .RegisterScenarios(scenario)
            .Run();
    }
}
